src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -242,10 +242,12 @@ private function handleException(\Exception $e, $request, $type)
 
         // the developer asked for a specific status code
         if ($response->headers->has('X-Status-Code')) {
+            @trigger_error(sprintf('Using the X-Status-Code header is deprecated, use %s::setAllowSuccessfulResponse() instead.', GetResponseForExceptionEvent::class), E_USER_DEPRECATED);
```

Comment:
They aren't in other deprecation warnings so I've followed the same pattern here